server/src/controllers/Blogs.js

```diff
@@ -2,61 +2,67 @@ const User = require("../models/User");
 const Blog = require("../models/Blog");
 
 module.exports = {
-  async allBlogs(_req, res) {
-    const blogs = await Blog.find();
-    res.status(200).json(blogs);
+
+  async allBlogs(_req, res, next) {
+    try{
+      const blogs = await Blog.find();
+      return res.status(200).json(blogs);
+    }catch(e){
+      return res.status(400).json({error:e.message});
+    }
   },
 
   async viewBlogById(req, res) {
-    const blogId = req.params.id;
-    const blog = await Blog.findById(blogId);
-    if (blog) {
-      res.status(200).json(blog);
-    } else {
-      res.status(404).json({
-        error: "The requested blog doesn't exist",
-      });
+    try{
+      const blogId = req.params.id;
+      const blog = await Blog.findById(blogId);
+      if (blog) {
+        return res.status(200).json(blog);
+      } else {
+        return res.status(404).json({
+          error: "The requested blog doesn't exist",
+        });
+      }
+    }catch(e){
+      return res.status(500).json({error:e.message})
     }
   },
 
-  async uploadBlog(req, res) {
-    // TODO: add isBlogAuthorized middleware
+  async uploadBlog(req, res, next) {
     try {
       // Assumed that req.body already has required fields
       const blog = await Blog.create(req.body);
-      res.status(201).json({
+      return res.status(201).json({
         id: blog._id,
       });
     } catch (error) {
       // console.log(error);
-      res.status(500).json({
-        error: error,
-      });
+      return res.status(500).json({message:error.message});
     }
   },
 
-  async editBlogById(req, res) {
+  async editBlogById(req, res, next) {
     // TODO: add isBlogAuthorized middleware
     try {
       // Assumed that req.body already has required fields
       const blog = await Blog.findByIdAndUpdate(req.params.id, req.body, {
         new: true,
       });
-      res.json({
+      return res.json({
         id: blog._id,
         blogTitle: blog.blogTitle,
       });
     } catch (error) {
-      res.status(400).json({
-        error: error,
+        return res.status(400).json({
+        error: error.message,
       });
     }
   },
-  async deleteBlogById(req, res) {
+  async deleteBlogById(req, res, next) {
     // TODO: add isBlogAuthorized middleware
     const blogId = req.params.id;
     const blog = await Blog.findById(blogId);
     await blog.remove();
-    res.status(204).json({});
+    return res.status(204).json({message:"Successfully deleted blog"});
```

Comment:
A status of 204 indicates No Content (see [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204)). It'd be better to remove the message and pass in an empty object here.